pkg/cmd/render/render.go

```diff
@@ -0,0 +1,196 @@
+
+	return cmd
+}
+
+func (r *renderOpts) AddFlags(fs *pflag.FlagSet) {
+	fs.StringVar(&r.ppInputFileDir, "performance-profile-dir-path", r.ppInputFileDir, "Input path for the Performance Profile manifest's directory.")
+	fs.StringVar(&r.assetsInDir, "asset-input-dir", components.AssetsDir, "Input path for the assets directory.")
+	fs.StringVar(&r.assetsOutDir, "asset-output-dir", r.assetsOutDir, "Output path for the rendered manifests.")
+	// environmet variables has precedence over standard input
+	r.readFlagsFromEnv()
+}
+
+func (r *renderOpts) readFlagsFromEnv() {
+	if ppDir := os.Getenv("PP_DIR"); len(ppDir) > 0 {
+		r.ppInputFileDir = ppDir
+	}
+
+	if assetInDir := os.Getenv("ASSET_IN_DIR"); len(assetInDir) > 0 {
+		r.assetsInDir = assetInDir
+	}
+
+	if assetsOutDir := os.Getenv("ASSET_OUT_DIR"); len(assetsOutDir) > 0 {
+		r.assetsOutDir = assetsOutDir
+	}
+}
+
+func (r *renderOpts) Validate() error {
+	if len(r.ppInputFileDir) == 0 {
+		return fmt.Errorf("performance-profile-file-path must be specified")
+	}
+
+	if len(r.assetsOutDir) == 0 {
+		return fmt.Errorf("asset-output-dir must be specified")
+	}
+
+	return nil
+}
+
+func (r *renderOpts) Run() error {
+	PPFiles := getPPList(r.ppInputFileDir)
+	if len(PPFiles) == 0 {
+		return fmt.Errorf("no valid performance-profiles found under: %v", r.ppInputFileDir)
+	}
+
+	for idx, pp := range PPFiles {
+		b, err := ioutil.ReadFile(filepath.Join(r.ppInputFileDir, pp))
+		if err != nil {
+			return err
+		}
+		components := make(manifestable)
+
+		profile := &performancev2.PerformanceProfile{}
+		err = yaml.Unmarshal(b, profile)
+		if err != nil {
+			return err
+		}
+		or := []metav1.OwnerReference{
+			{
+				Kind: profile.Kind,
+				Name: profile.Name,
+			},
+		}
+
+		mc, err := machineconfig.New(r.assetsInDir, profile)
+		if err != nil {
+			return err
+		}
+		components[mc.Kind] = mc
+
+		kc, err := kubeletconfig.New(profile)
+		if err != nil {
+			return err
+		}
+		components[kc.Kind] = kc
+
+		performanceTuned, err := tuned.NewNodePerformance(r.assetsInDir, profile)
+		if err != nil {
+			return err
+		}
+		components[performanceTuned.Kind] = performanceTuned
+
+		runtimeClass := runtimeclass.New(profile, machineconfig.HighPerformanceRuntime)
+		components[runtimeClass.Kind] = runtimeClass
+
+		for name, component := range components {
+			component.SetOwnerReferences(or)
+			b, err := yaml.Marshal(component)
+			if err != nil {
+				return err
+			}
+
+			fileName := fmt.Sprintf("%s-%d.yaml", name, idx)
+			err = ioutil.WriteFile(filepath.Join(r.assetsOutDir, fileName), b, 0644)
+			if err != nil {
+				return err
+			}
+		}
+	}
+
+	return nil
+}
+
+// getPPList return a list of performance-profile manifests under dirPath
+func getPPList(dirPath string) []string {
+	files, err := ioutil.ReadDir(dirPath)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	// some regex to look for performance-profile manifests in directory's path
+	reg, err := regexp.Compile(`[P,p].*[-,_]*[P,p].*\.y[a]*ml`)
```

Comment:
why is this needed?

Reply:
I don't know the performance profile name and there might be other files in the directory. so i'm looking for yaml's file that matches this regex

Reply:
How others operator detect the input file? Do they have a separate command line flag?
Another option could be just decode all the YAMLs in the input directory, and look for a performance profile. I actually like this more than looking for specific names.